### ISSUE TITLE
Idea for pytext

### ISSUE THREAD

**Issue by Anonymous6598** [lorenzolpandolfo/pytext#19]
Hi @lorenzolpandolfo!

How are you? I have good news for your project - Acascape has released a new module for customtkinter - CTkCodeBox. It's a new widget, which is like a chlorophyll widget, but supports all programming language by default and has a lot of cool features. It will be good for your project because it will optimize your program by reducing the number of lines in your program and will make it more useable, just like Vim. I'll send you a link to his repo later. So what do you think?

@Anonymous6598

**Comment by Anonymous6598** [comment]
Link for repo and docs: https://github.com/Akascape/CTkCodeBox

**Comment by lorenzolpandolfo** [comment]
Hello @Anonymous6598 !
Thank you for your recommendation. Pytext is no longer using Customtkinter, I moved all project to Tkinter to have a bigger control of theming and to have a better performance.
I implemented syntax highlight (it is working just for Python files at moment) and I made some optimization algorithms, since It can load very big documents (5000+ lines) and syntax highlight it. At moment, it has a very good performance, but with older laptops it can be a little laggy.
I checked the CTkCodeBox code and I didn't find any optimization algorithm, it just syntax highlight all document using Pygments, and I really think that It may crash with bigger files.
At moment, I made a algorithm that just syntax highlight the text visible in the screen. All other lines' tokens are deleted from memory so It can perform well.
Thank you for your message! I will check CTkCodeBox with more time when possible.
Best wishes! 😊
